logo.js

```diff
@@ -2443,11 +2622,11 @@ function LogoInterpreter(turtle, stream, savehook)
   });
 
   def("stop", function() {
-    throw new Output();
+    return Promise.reject({special: "stop"});
```

Comment:
Why can't this simply be `return Promise.reject(new Output());` ?

But more importantly, if the procedure invoker does the right thing, this could still be `throw new Output();`

```js
// assume proc and inputs
var p = new Promise(function(resolve) {
  resolve(proc.apply(self, inputs));
});
```
* If proc returns a non-Promise, p will be a Promise resolved to that value
* If proc returns a Promise, p will be a Promise that resolves to the same value
* If proc throws, p will be a rejected Promise